### PR TITLE
news-block test cases

### PR DESCRIPTION
**Pull request by cod17828**
PR includes

implementation of non-logged in and logged in scenarios of news-block
config changes (run dx nala tests on feature branch from which pr is created, or on partners.stage.adobe.com)

### CODE REVIEW

File: tests/dx/news.test.js [hunk 1]
```diff
@@ -6,90 +6,165 @@ const News = require('../../features/dx/news.spec.js');
...
   test.beforeEach(async ({ page }) => {
     newsPage = new NewsPage(page);
   });
+
+  function getEmail(imsCredentials, partnerLevel) {
+    return imsCredentials.split(partnerLevel)[1].split(';')[0];
+  }
+
+  async function signIn(page, partnerLevel) {
+    await newsPage.IMSEmailPage.waitFor({ state: 'visible', timeout: 5000 });
```
Comment (Dli3) [inline tests/dx/news.test.js:47]: You probably don't need this wait. Your emailField.fill() should autowait already. Please see https://playwright.dev/docs/actionability.

File: tests/dx/news.test.js [hunk 2]
```diff
@@ -6,90 +6,165 @@ const News = require('../../features/dx/news.spec.js');
...
+    await newsPage.emailField.fill(email);
+    await newsPage.emailPageContinueButton.click();
+    await page.waitForLoadState('domcontentloaded');
+    await newsPage.IMSPasswordPage.waitFor({ state: 'visible', timeout: 10000 });
+    await newsPage.passwordField.fill(process.env.IMS_PASS);
+    await newsPage.passwordPageContinueButton.click();
+    await page.waitForLoadState('domcontentloaded');
+  }
+
+  async function runLoginTest({
```
Comment (Dli3) [inline tests/dx/news.test.js:58]: You don't want to add these functions to your page object class?

File: tests/dx/news.test.js [hunk 3]
```diff
@@ -135,7 +210,160 @@ test.describe('Validate news block', () => {
...
       const resultAfterClearingAllFilters = await newsPage.resultNumber.textContent();
-      await expect(parseInt(resultAfterClearingAllFilters.split(' ')[0], 10)).toBeGreaterThan(6);
+      await expect(parseInt(resultAfterClearingAllFilters.split(' ')[0], 10)).toBe(9);
+    });
+  });
+
+  test(`${features[3].name},${features[3].tags}`, async ({ page, baseURL }) => {
+    await test.step('Go to News page', async () => {
+      await page.goto(`${baseURL}${features[3].path}`);
+      await newsPage.firstCardDate.waitFor({ state: 'visible', timeout: 10000 });
```
Comment (Dli3) [inline tests/dx/news.test.js:220]: You probably don't need this wait. Your emailField.fill() should autowait already. Please see https://playwright.dev/docs/actionability.

File: tests/dx/news.test.js [hunk 4]
```diff
@@ -135,7 +135,174 @@ test.describe('Validate news block', () => {
...
+      cardPartnerLevelAbove: 'Automation regression news card spp gold no1',
+    },
+    {
+      id: 8,
+      partnerLevel: 'spp-bronze:',
+      resultTotal: 11,
+      cardPartnerLevel: 'Automation regression news card SPP Bronze no1',
+      cardPartnerLevelAbove: 'Automation regression news card spp silver no1',
+    },
+    {
```
Comment (Dli3) [inline tests/dx/news.test.js:245]: I think you can add this to your spec file. Maybe add a data key. 
For example:
  ```
  {
      tcid: '8',
      name: '@login-news-page-silver-user',
      path: '/solutionpartners/drafts/automation/regression/partner-news',
      tags: '@dx-news @regression @login',
      data: {
      partnerLevel: 'spp-bronze:',
      resultTotal: 11,
      cardPartnerLevel: 'Automation regression news card SPP Bronze no1',
      cardPartnerLevelAbove: 'Automation regression news card spp silver no1',
}
    },
```

**Review by Dli3 — APPROVED** [review]

File: tests/dx/news.test.js (hunk 4 above)
Comment (skumar09) [reply tests/dx/news.test.js:245]: Yes, pls include the test case, and test data in `.spec `file

**Review by skumar09 — CHANGES_REQUESTED** [review]

File: tests/dx/news.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (cod17828) [reply tests/dx/news.test.js]: @skumar09 why would it not pick GitHub .env variables?

**Review by JackySun9 — APPROVED** [review]

**Review by skumar09 — APPROVED** [review]

**Pull request merged**